api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -288,4 +288,10 @@ message Resource {
   // testing where the fault injection should be terminated in the event that Envoy loses contact
   // with the management server.
   google.protobuf.Duration ttl = 6;
+
+  // If true, xDS proxies may not cache this resource.
+  // Note that this does not apply to clients other than xDS proxies, which must cache resources
+  // for their own use, regardless of the value of this field.
+  // [#not-implemented-hide:]
+  bool do_not_cache = 7;
```

Comment:
LGTM, but maybe make this a `CacheControl` message, in case we need something more here in the future..

Reply:
+1 to a wrapper message otherwise LGTM

Reply:
Done.